pkg/config/spec.go

```diff
@@ -276,28 +276,49 @@ func NewUpgradeSpec(cfg *Config) (*v1.UpgradeSpec, error) {
 			recMnt = constants.TransitionDir
 		}
 
+		upgradeRecoverySystemUri, err := cfg.Query("upgrade.\"recovery-system\".uri")
```

Comment:
I'm trying to fix the tests currently. I think this should be `install.recovery-system.uri` but I'm not sure yet. On it.